ui/raidboss/data/03-hw/trial/sophia-ex.js

```diff
@@ -6,7 +6,409 @@
+          '-55.06373.523648': 3,
+          '54.99073.387837': 4,
+          '54.986999.576593': 5,
+          '54.9907-3.50686': 6,
+          '54.99068-10.14043': 7,
+        };
+        // Horrible, awful string manipulation, whyyy, Javascript?
+        const seqStart = parseInt(matches.sourceId, 16) - scaleMap[matches.x + matches.y];
+        for (let i = 0; i < 8; i++) {
+          data.scaleSophias = data.scaleSophias || [];
+          data.scaleSophias.push((seqStart + i).toString(16).toUpperCase());
+        }
+      },
+    },
+    {
+      // We collect tethers here for later use on Quasar calls.
+      // Blue Quasars (19A9) weigh 3 units, while orange ones (19A8) weigh 1.
+      // Meteor Quasars happen only when we have a 3/1 split of tethers (balanced).
+      // If the tether split is not 3/1, it will always be one of these possibilities:
+      // 1/1, 2/2, 3/3, 2/1, 3/2, 4/2, 4/3
+      // If the difference of the sum of weights on each side is 1, the tilt will be soft.
+      // Otherwise it will be hard.
+      // There will always be exactly one blue Quasar, unless the split is 4/2.
+      id: 'SophiaEX Quasar Tether Collect',
+      netRegex: NetRegexes.tether({ id: '0011' }),
+      condition: function(data) {
+        // We shouldn't run this while Aion Teleos mechanics are active.
+        return !data.clonesActive && data.scaleSophias;
+      },
+      run: function(data, matches) {
+        data.quasarTethers = data.quasarTethers || [];
+        data.quasarTethers.push(matches.sourceId);
+      },
+    },
+    {
+      id: 'SophiaEX Tilt Via Tether',
+      netRegex: NetRegexes.tether({ id: '0011', capture: false }),
+      condition: function(data) {
+        // No platform tilts if clones are up, and no tilts if it's the first one.
+        return !data.clonesActive && data.scaleSophias;
+      },
+      // We let the storage triggers catch up before calling.
+      delaySeconds: .5,
+      durationSeconds: 12, // Ensuring that forgetful people aren't forgotten.
+      suppressSeconds: 5,
+      alertText: function(data) {
+        // Tethers are ordered with all East tethers first. This *doesn't* mean that the East
+        // or West tethers are themselves in order within their half!
+        // The eight scale entities are listed in the data object, with West at indices 0-3,
+        // under data.scaleSophias.
+        let safeDir = 0;
+        // If there's a side with more tethers, we know for sure that's the safe side.
+        // This will give us the tilt direction for all but the 1/1, 2/2, and 3/3 cases.
+        // The safe side is represented here by whether safeDir is positive or negative.
+        // (West/negative, East/positive.)
+        for (let i = 0; i < data.quasarTethers.length; i++)
+          safeDir += data.scaleSophias.indexOf(data.quasarTethers[i]) < 4 ? -1 : 1;
+        if (safeDir == 0) {
+          // If it's the 1/1, 2/2, or 3/3 case, we sadly don't have enough information.
```

Comment:
From the time this condition is met to the time the Quasar starts casting is about 3.2 seconds. (I've removed the delay on the "call from the Quasar cast" trigger since there shouldn't actually be a need for it.) Given that this is no slower than a player would be able to react on their own, I think it's probably fine as it is.

Reply:
Thanks, that seems reasonable!